app/RouteHandlers/SofortNotificationHandler.php

```diff
@@ -16,16 +17,28 @@ class SofortNotificationHandler {
 
 	private $ffFactory;
 
+	/**
+	 * @var Request
+	 */
+	private $request;
+
 	public function __construct( FunFunFactory $ffFactory ) {
 		$this->ffFactory = $ffFactory;
 	}
 
 	public function handle( Request $request ): Response {
-		$useCase = $this->ffFactory->newHandleSofortPaymentNotificationUseCase( $request->query->get( 'updateToken' ) );
+		$this->request = $request;
+
+		$useCaseRequest = $this->newUseCaseRequest();
+
+		if ( $useCaseRequest === null ) {
+			$this->logWebRequest( [ 'message' => 'Bad notification time' ], LogLevel::ERROR );
+			return new Response( 'Bad request', Response::HTTP_BAD_REQUEST );
```

Comment:
In the (admittedly unlikely) case that we have another validation in `newUseCaseRequest`, this will lead to a misleading error message. Could you use an exception and try...catch instead?